src/node_rc.cc

```diff
@@ -0,0 +1,198 @@
+}
+
+std::optional<std::string> ConfigReader::GetDataFromArgs(
+    const std::vector<std::string>& args) {
+  constexpr std::string_view flag = "--experimental-config-file";
+
+  for (auto it = args.begin(); it != args.end(); ++it) {
+    if (*it == flag) {
+      // Case: "--experimental-config-file foo"
+      if (auto next = std::next(it); next != args.end()) {
+        return *next;
+      }
+    } else if (it->starts_with(flag)) {
+      // Case: "--experimental-config-file=foo"
+      if (it->size() > flag.size() && (*it)[flag.size()] == '=') {
+        return it->substr(flag.size() + 1);
+      }
+    }
+  }
+
+  return std::nullopt;
+}
+
+ConfigReader::ParseResult ConfigReader::ParseConfig(
+    const std::string& config_path) {
+  std::string file_content;
+  // Read the configuration file
+  int r = ReadFileSync(&file_content, config_path.c_str());
+  if (r != 0) {
+    const char* err = uv_strerror(r);
+    FPrintF(
+        stderr, "Cannot read configuration from %s: %s\n", config_path, err);
+    return ParseResult::FileError;
+  }
+
+  // Parse the configuration file
+  simdjson::ondemand::parser json_parser;
+  simdjson::ondemand::document document;
+  if (json_parser.iterate(file_content).get(document)) {
+    FPrintF(stderr, "Can't parse %s\n", config_path.c_str());
+    return ParseResult::InvalidContent;
+  }
+
+  simdjson::ondemand::object main_object;
+  // If document is not an object, throw an error.
+  if (auto root_error = document.get_object().get(main_object)) {
+    if (root_error == simdjson::error_code::INCORRECT_TYPE) {
+      FPrintF(stderr,
+              "Root value unexpected not an object for %s\n\n",
+              config_path.c_str());
+    } else {
+      FPrintF(stderr, "Can't parse %s\n", config_path.c_str());
+    }
+    return ParseResult::InvalidContent;
+  }
+
+  ConfigReader::Config config;
+  simdjson::ondemand::value ondemand_value;
+  simdjson::ondemand::raw_json_string key;
+
+  for (auto field : main_object) {
+    if (field.key().get(key) || field.value().get(ondemand_value)) {
+      return ParseResult::InvalidContent;
+    }
+    if (key == "experimental") {
+      simdjson::ondemand::object experimental_object;
+      if (ondemand_value.get_object().get(experimental_object)) {
+        FPrintF(stderr, "Invalid value for experimental object\n");
+        return ParseResult::InvalidContent;
+      }
+      auto result = ParseExperimentalObject(&experimental_object, &config);
+      if (result != ParseResult::Valid) {
+        return result;
+      }
+    }
+
+    if (key == "test") {
+      simdjson::ondemand::object test_object;
+      if (ondemand_value.get_object().get(test_object)) {
+        FPrintF(stderr, "Invalid value for test object\n");
+        return ParseResult::InvalidContent;
+      }
+      auto result = ParseTestObject(&test_object, &config);
+      if (result != ParseResult::Valid) {
+        return result;
+      }
+    }
+
+    if (key == "import") {
```

Comment:
Please add else if to each non first statement

```suggestion
    else if (key == "import") {
```